static/js/components/ProgramSelector.js

```diff
@@ -0,0 +1,119 @@
+    setEnrollDialogVisibility:   (open: boolean) => void,
+    setEnrollSelectedProgram:    (programId: number) => void,
+    setProgramSelectorOpen:      (open: boolean) => void,
+  };
+
+  handleProgramChange = (event: any): void => {
+    const { enrollments, setCurrentProgramEnrollment } = this.props;
+    const value = parseInt(event.target.value);
+
+    const enrollment = enrollments.programEnrollments.find(enrollment => enrollment.id === value);
+    setCurrentProgramEnrollment(enrollment);
+  };
+
+  handleOpenSelect = () => {
+    const { programSelectorOpen, setProgramSelectorOpen } = this.props;
+    setProgramSelectorOpen(!programSelectorOpen);
+  };
+
+  selectEnrollment = (enrollment: ProgramEnrollment): void => {
+    const { setCurrentProgramEnrollment, setProgramSelectorOpen } = this.props;
+    setCurrentProgramEnrollment(enrollment);
+    setProgramSelectorOpen(false);
+  };
+
+  showNewEnrollmentDialog = () => {
+    let { setEnrollDialogVisibility } = this.props;
+    setEnrollDialogVisibility(true);
+  };
+
+  render() {
+    let {
+      enrollments: { programEnrollments },
+      programSelectorOpen,
+      dashboard: { programs },
+    } = this.props;
+    let { currentProgramEnrollment } = this.props;
+    let currentId;
+    if (currentProgramEnrollment !== null) {
+      currentId = currentProgramEnrollment.id;
+    }
+
+    programEnrollments = _.sortBy(programEnrollments, 'title');
+    if (programEnrollments.length === 0) {
+      return <div className="program-selector" />;
+    }
+
+    let selected = programEnrollments.find(enrollment => enrollment.id === currentId);
+    let unselected = programEnrollments.filter(enrollment => enrollment.id !== currentId);
+    let options = unselected.map(enrollment => (
+      <div
+        className="option"
+        key={enrollment.id}
+        onClick={() => this.selectEnrollment(enrollment)}
+      >
+        {enrollment.title}
+      </div>
+    ));
+
+    let enrollmentLookup = new Map(programEnrollments.map(enrollment => [enrollment.id, null]));
+    let unenrolledPrograms = programs.filter(program => !enrollmentLookup.has(program.id));
+    unenrolledPrograms = _.sortBy(unenrolledPrograms, 'title');
+
+    let enrollOption;
+    if (unenrolledPrograms.length > 0) {
+      enrollOption = <div
+        className="option enroll-new-program"
+        key="enroll-new-program"
+        onClick={() => this.showNewEnrollmentDialog()}
+      >
+        Enroll in a new program
+      </div>;
+    }
+
+    let select = <div className="select-container">
+      <div className="select">
+        <div className="selected-option" onClick={this.handleOpenSelect}>
+          {selected.title} <i className="material-icons">arrow_drop_down</i>
+        </div>
+        <div className={`select-dropdown ${programSelectorOpen ? 'open' : ''}`}>
+          {options}
+          {enrollOption}
+        </div>
+      </div>
+    </div>;
```

Comment:
Do you mind if I postpone this for another PR? The `<select>` does not seem viable and I can't think of a good way to refactor this at the moment